### PR TITLE
Add blog

### LINKED ISSUE #1636: Refactor blog

### PR DESCRIPTION
**Pull request by IvanDanilov66**
Fixes #1597 
Fixes #1636 

We are using styled components only on Slider but I think that it should not be a problem, because we have to redo the whole Slider for only that. (The lint problem is fixed)

### CODE REVIEW

**Comment by IvanDanilov66** [comment]
We will be making it without styled components this week, so we will push on this branch. You can merge this now.

File: package.json [hunk 1]
```diff
@@ -72,9 +72,11 @@
     "@types/leaflet": "^1.5.19",
     "@types/lodash-es": "^4.17.3",
     "@types/qrcode": "^1.3.5",
+    "@types/react": "^17.0.1",
```
Comment (angle943) [inline package.json:75]: why did you downgrade @types/react?

Comment (IvanDanilov66) [reply package.json:75]: Maybe because you updated it in the meantime and I was using the old one.

Comment (IvanDanilov66) [reply package.json]: I have updated it to 17.0.3 (the newest version)

**Review by jamessspanggg — CHANGES_REQUESTED** [review]
There are a lot of console errors with your addition:

validateDomNesting error:
Screen Shot 2021-03-24 at 8 35 08 PM


error when focused on subscribe to email input:
Screen Shot 2021-03-24 at 8 36 14 PM

error when visiting article page from main blog page:
Screen Shot 2021-03-24 at 8 35 18 PM

key error:
Screen Shot 2021-03-24 at 8 35 29 PM

**Comment by jamessspanggg** [comment]
> We are using styled components only on Slider but I think that it should not be a problem, because we have to redo the whole Slider for only that. 

I think there's no need to use `styled-components` as there are only two of your components that uses it, the rest uses a separate `scss` file. 

@angle943  @thesanjeevsharma thoughts?

**Comment by thesanjeevsharma** [comment]
Yeah, as the author mentioned he's gonna refactor it and remove styled-components this week. Should we wait until then? or we can allow this and later on merge the PR with the refactor?

IMO, it won't take much time to refactor so let's keep this open and wait for the refactor to be done first.

**Comment by IvanDanilov66** [comment]
All the errors are fixed @jamessspanggg can you please check now .

File: src/containers/TopNav/TopNavDesktopItems/index.tsx [hunk 2]
```diff
@@ -59,6 +59,13 @@ const morePopoverItems: TopNavPopoverItemType[] = [
     title: 'Project Proposals',
     to: '/project-proposals/overview',
   },
+  {
+    description: 'Stay updated on crypto and thenewboston',
+    iconType: IconType.textBox,
+    // isExternal: true,
```
Comment (angle943) [inline src/containers/TopNav/TopNavDesktopItems/index.tsx:65]: remove this comment

File: src/containers/TopNav/TopNavDesktopItems/index.tsx [hunk 3]
```diff
@@ -59,6 +59,13 @@ const morePopoverItems: TopNavPopoverItemType[] = [
...
+    title: 'Blog',
+    to: '/blog', // 'https://thenewboston.blog/'
```
Comment (angle943) [inline src/containers/TopNav/TopNavDesktopItems/index.tsx:67]: i don't think we need this comment

File: src/containers/Blog/Subscribe/index.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  const validationSchema = yup.object().shape({
+    email: yup.string().email().required('Email is required'),
+  });
+
+  const renderSocialMediaLinks = () =>
+    [SocialMedia.linkedin, SocialMedia.facebook, SocialMedia.twitter, SocialMedia.instagram].map((website) => (
+      <SocialMediaIcon className="Subscribe__SocialMediaLink" iconSize={28} key={website} website={website} />
+    ));
+  return (
+    <div className="Subscribe">
```
Comment (angle943) [inline src/containers/Blog/Subscribe/index.tsx:20]: we have four nested `div` here, can we simplify this?

File: src/containers/Blog/Subscribe/Subscribe.scss [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+    justify-content: space-between;
+    margin-top: 1.5rem;
+    width: 200px;
+  }
+
+  &__SocialMediaLink {
+    color: $gray-900;
+  }
+
+  //   Forms
```
Comment (angle943) [inline src/containers/Blog/Subscribe/Subscribe.scss:105]: comment not needed

File: src/containers/Blog/Subscribe/Subscribe.scss [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+  }
+
+  &__content-container {
+    margin-bottom: 5rem;
+    max-width: $summary-width + $summary-mr + $comparison-cards-container + $content-container-padding-lr * 2;
+    padding: 64px;
+    width: 100%;
+
+    @media (max-width: 768px) {
+      //    margin-bottom: 3rem;
```
Comment (angle943) [inline src/containers/Blog/Subscribe/Subscribe.scss:40]: remove comments

File: src/containers/Blog/RecentlyPosted/index.tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  zoomFactor: 1,
+};
+
+const RecentlyPosted: FC<ArticlesProps> = ({articles}) => {
+  return (
+    <div className="RecentlyPosted">
+      <div className="RecentlyPosted__content-container">
+        <h3 className="RecentlyPosted__header">Recently Posted</h3>
+        <div className="RecentlyPosted__article">
+          {/* eslint-disable-next-line react/jsx-props-no-spreading */}
```
Comment (angle943) [inline src/containers/Blog/RecentlyPosted/index.tsx:24]: this line is not needed, as you have a file-wide eslint-disable on top of this file

File: src/containers/Blog/RecentlyPosted/index.tsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  maxVisibleSlides: 3,
+  pageTransition: 850,
+  slideMargin: 5,
+  zoomFactor: 1,
+};
+
+const RecentlyPosted: FC<ArticlesProps> = ({articles}) => {
+  return (
+    <div className="RecentlyPosted">
+      <div className="RecentlyPosted__content-container">
```
Comment (angle943) [inline src/containers/Blog/RecentlyPosted/index.tsx:21]: are these two `div` necessary? can we make them into 1?

File: src/containers/Blog/RecentlyPosted/Article.tsx [hunk 9]
```diff
@@ -0,0 +1,47 @@
+import React, {FC} from 'react';
+import {Link} from 'react-router-dom';
+import {Article as ArticleType} from 'types/blogs';
+import {slugify} from 'utils/urls';
+
+import './RecentlyPosted.scss';
+
+// {
```
Comment (angle943) [inline src/containers/Blog/RecentlyPosted/Article.tsx:8]: what is this comment for?

File: src/containers/Blog/BlogHero/PopularArticles/PopularArticles.scss [hunk 10]
```diff
@@ -0,0 +1,130 @@
+$comparison-cards-container: 690px;
+$content-container-padding-lr: 64px;
+$summary-mr: 70px;
+$summary-width: 420px;
+$gray-500: #697386;
+$gray-100: #e3e8ee;
+
+.PopularArticles {
+  // align-items: center;
```
Comment (angle943) [inline src/containers/Blog/BlogHero/PopularArticles/PopularArticles.scss:9]: let's remove these commented out styles

File: src/containers/Blog/BlogHero/PopularArticles/Article.tsx [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+}
+const Article: FC<ComponentProps> = ({article}) => {
+  return (
+    <div className="PopularArticles__article-wrapper">
+      <div className="PopularArticles__article-banner">
+        <Link to={`/blog/${encodeURIComponent(slugify(article.title, ' ', '-'))}`}>
+          <img src={article.banner} alt="img" />
+        </Link>
+      </div>
+      {/* <div className="PopularArticles__article-banner" /> */}
```
Comment (angle943) [inline src/containers/Blog/BlogHero/PopularArticles/Article.tsx:26]: what's this?

File: src/containers/Blog/BlogHero/ActiveArticle/index.tsx [hunk 12]
```diff
@@ -0,0 +1,46 @@
+/* eslint-disable no-console */
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
Comment (angle943) [inline src/containers/Blog/BlogHero/ActiveArticle/index.tsx:2]: we should remove these two eslint-disable

File: src/containers/Blog/Blog.scss [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+    &__Step {
+      //margin: 0 0 48px 0;
+      margin: 0;
+      max-width: 430px;
+    }
+  }
+
+  @media (max-width: 480px) {
+    &__Step {
+      //margin: 0 32px 32px;
```
Comment (angle943) [inline src/containers/Blog/Blog.scss:51]: comment

File: src/containers/Blog/Blog.scss [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+  }
+
+  @media (max-width: 1080px) {
+    &__content-container {
+      align-items: center;
+      flex-direction: row;
+    }
+
+    &__Step {
+      //margin: 0 0 48px 0;
```
Comment (angle943) [inline src/containers/Blog/Blog.scss:43]: comment

File: src/containers/Blog/Article/index.tsx [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+                </div>
+                {/* Share btn */}
+                <div style={{position: 'relative'}}>
+                  <Icon
+                    size={25}
+                    icon={IconType.shareCircle}
+                    className="Article__SocialMediaShareBtn"
+                    onClick={() => toggleSocialVisible()}
+                  />
+                  {/* <div className="Article__likes-mobile">
```
Comment (angle943) [inline src/containers/Blog/Article/index.tsx:75]: i think we can remove this

File: src/containers/Blog/Article/index.tsx [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+import React, {FC, useMemo, useState} from 'react';
+import {useParams} from 'react-router-dom';
+import {A, Icon, IconType} from 'components';
+import {SocialMedia} from 'types/social-media';
+
+import './Article.scss';
+import {slugify} from 'utils/urls';
+import {getArticleByTitle} from 'utils/data';
+
+// {
```
Comment (angle943) [inline src/containers/Blog/Article/index.tsx:11]: what is this comment for?

File: src/containers/Blog/Article/Article.scss [hunk 17]
```diff
@@ -0,0 +1,218 @@
...
+    opacity: 0;
+    /* width: 100%; */
+    padding: 0.5rem 0;
+    position: absolute;
+    right: 0;
+    transform: translateY(10px);
+    transition: 0.5s;
+    z-index: 100;
+
+    // ::before {
```
Comment (angle943) [inline src/containers/Blog/Article/Article.scss:155]: commented out style

File: src/containers/Blog/Article/Article.scss [hunk 18]
```diff
@@ -0,0 +1,218 @@
...
+    }
+  }
+
+  &__SocialMediaLinkLabel {
+    font-size: 13px;
+    margin: 0 0.5rem;
+  }
+
+  &__social-mobile {
+    // padding-top: 0;
```
Comment (angle943) [inline src/containers/Blog/Article/Article.scss:139]: commented out style here, plus width: 100% couple lines below

File: src/containers/Blog/Article/Article.scss [hunk 19]
```diff
@@ -0,0 +1,218 @@
...
+  }
+
+  // Social Icon
+  &__Socials__wrapper {
+    display: none;
+
+    @media (min-width: 1200px) {
+      display: block;
+      position: fixed;
+      // left: 20px;
```
Comment (angle943) [inline src/containers/Blog/Article/Article.scss:103]: comment

File: src/containers/Blog/Article/Article.scss [hunk 20]
```diff
@@ -0,0 +1,218 @@
...
+    margin: 0;
+  }
+
+  &__time {
+    margin-left: 1rem;
+  }
+
+  //   Banner
+  &__banner {
+    // background-color: $gray-500;
```
Comment (angle943) [inline src/containers/Blog/Article/Article.scss:70]: comment

**Comment by angle943** [comment]
@IvanDanilov66 @jamessspanggg let's remove styled components before merging it in

**Comment by angle943** [comment]
@IvanDanilov66 are there any diffence between the contents inside `components/FormComponents` folder, and the one you created: `components/FormComponentsBlog`?

**Comment by jamessspanggg** [comment]
Updated the blog to be more responsive, as well as remove some unwanted styles as it seems like there are a lot of styles being copy and pasted that are unused.

Before:

https://user-images.githubusercontent.com/32864116/112849294-80ea2c00-90db-11eb-9899-41f8d7b0d603.mp4


After:

https://user-images.githubusercontent.com/32864116/112850314-77ad8f00-90dc-11eb-9c04-53fa83087a18.mp4

**Comment by angle943** [comment]
Closing this issue, as we have discussed about an alternative way to presenting blogs.

@IvanDanilov66 what is your account #?

**Comment by IvanDanilov66** [comment]
> Closing this issue, as we have discussed about an alternative way to presenting blogs.
> 
> @IvanDanilov66 what is your account #?

f1d85cecf85d7da38401dc58fabf1aa4294c881bb9d2c47642d1b460017f06cd

**Pull request closed without merging**